### PR TITLE
Documentation for UOs and some time schemes

### PR DESCRIPTION
**Pull request by GiudGiud**

### CODE REVIEW

**Comment by GiudGiud** [comment]
need to fix all the butcher tableaus

**Comment by GiudGiud** [comment]
struggled with formatting for a long time, but should be ready for review now

File: test/tests/userobjects/Terminator/terminator_pass.i [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  ny = 6
+  xmin = -15.0
+  xmax = 15.0
+  ymin = -3.0
+  ymax = 3.0
+  elem_type = QUAD4
+[]
+
+[Variables]
+  [./c]
```
Comment (joshuahansel) [inline test/tests/userobjects/Terminator/terminator_pass.i:24]: Get these "./" out of here!

File: test/tests/userobjects/Terminator/terminator_pass.i [hunk 2]
```diff
@@ -0,0 +1,81 @@
+###########################################################
+# This is a test of the UserObject System. The
+# Terminator UserObject executes independently after
+# each solve and can terminate the solve early due to
+# user-defined criteria. (Type: GeneralUserObject)
+#
+# @Requirement F6.40
```
Comment (joshuahansel) [inline test/tests/userobjects/Terminator/terminator_pass.i:7]: What is this? Is this something we still do or copy-paste error?

File: framework/doc/content/source/timeintegrators/AStableDirk4.md [hunk 3]
```diff
@@ -5,28 +5,30 @@
...
-        | $1/(24*(1/2-\gamma))$   $1 - 1/(12*(1/2-\gamma)^2)$  $1/(24*(1/2-\gamma)^2)$
+| $c_i$ | $a_{i1}$ | $a_{i2}$ | $a_{i3}$ | $a_{i4}$ |
+| - | - | - | - | - |
+| $\gamma$   | $\gamma$        | 0               | 0 |
+| 1/2        | $1/2-\gamma$    | $\gamma$        | 0 |
+| $1-\gamma$ | $2\gamma$        | $1-4\gamma$    | $\gamma$ |
+| $b_{j}$ | $\dfrac{1}{24(1/2-\gamma)}$    | $1 - \dfrac{1}{12(1/2-\gamma)^2}$ | $\dfrac{1}{24(1/2-\gamma)^2}$ |
 
-where $\gamma = 1/2 + sqrt(3)/3 * cos(pi/18) ~ 1.06857902130162881$
+where $\gamma = 1/2 + \dfrac{\sqrt(3)}{3} \cos( \dfrac{\pi}{18}) \approx 1.06857902130162881$
```
Comment (joshuahansel) [inline framework/doc/content/source/timeintegrators/AStableDirk4.md:15]: `\sqrt{3}`

File: framework/doc/content/source/timeintegrators/LStableDirk3.md [hunk 4]
```diff
@@ -5,25 +5,25 @@
...
+| $c_i$ | $a_{i1}$ | $a_{i2}$ | $a_{i3}$ | $a_{i4}$ |
+| - | - | - | - | - |
+| $\gamma$       | $\gamma$ |
+| $(1+\gamma)/2$ | $(1-\gamma)/2$ | $\gamma$ |
+| 1              | $(1/4)(-6\gamma^2 + 16\gamma - 1)$ | $(1/4)(6\gamma^2 - 20\gamma + 5)$ | $\gamma$ |
+| $b_j$ | $(1/4)(-6\gamma^2 + 16\gamma - 1)$ | $(1/4)(6\gamma^2 - 20 \gamma + 5)$ | $\gamma$ |
 
-where $\gamma = -sqrt(2)*cos(atan(sqrt(2)/4)/3)/2 + sqrt(6)*sin(atan(sqrt(2)/4)/3)/2 + 1  ~
-.435866521508459$
+where $\gamma = -\sqrt(2) \cos(atan(\sqrt(2)/4)/3)/2 + \sqrt(6) \sin(atan(\sqrt(2)/4)/3)/2 + 1  \approx 0.435866521508459$
```
Comment (joshuahansel) [inline framework/doc/content/source/timeintegrators/LStableDirk3.md:15]: \sqrt{}

File: framework/doc/content/source/timeintegrators/Ralston.md [hunk 5]
```diff
@@ -3,8 +3,8 @@
 !syntax description /Executioner/TimeIntegrator/Ralston
 
 Ralston's time integration method is second order accurate in time. It is a two step explicit
-method and a special case of the 2nd order Runge-Kutta method. It's obtained through an error minimization
-process and has been shown to outperform other 2nd order explicit Runge Kunta methods [!cite](ralston1962).
+method and a special case of the 2nd order Runge-Kutta method. It is obtained through an error minimization
+process and has been shown to outperform other 2nd order explicit Runge Kunta methods, see [!cite](ralston1962).
```
Comment (joshuahansel) [inline framework/doc/content/source/timeintegrators/Ralston.md:7]: - 2nd-order
- Runge-Kutta
- ", see" -> "; see" (it's a run-on)

**Review by joshuahansel — CHANGES_REQUESTED** [review]
Mainly commas, hyphens, and formatting issues, some just preference. I think we need to do something else for the Butcher tableau. Having "b_j" in the "c_i" column is confusing. How about we do it with a LaTeX equation so that we can render it in a more traditional manner?

Also (and mainly this is only personal preference) it's easier for me to review if when you fix something in earlier commits, you squash those commits together. This is because I review commit by commit, and sometimes you fix something I've already commented on. Maybe I just need to review the entire PR as one...

**Comment by GiudGiud** [comment]
Tableau in equation is genius. I ll do that

File: framework/doc/content/source/userobject/LayeredAverage.md [hunk 6]
```diff
@@ -1,33 +1,68 @@
...
-documentation clear for users.
+!syntax description /UserObjects/LayeredAverage
 
-```markdown
-# LayeredAverage
+## How to define the layers
 
-!syntax description /UserObjects/LayeredAverage
+The layer definition is handled by the `LayeredBase` class. It is summarized here.
+One or multiple layers are defined by their direction (normal or axis), their number
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply framework/doc/content/source/userobject/LayeredAverage.md:8]: ok let me try guessing what you want here lol

File: framework/doc/content/source/timeintegrators/CrankNicolson.md [hunk 7]
```diff
@@ -1,33 +1,22 @@
...
-MooseObject; however, what is contained is ultimately determined by what is necessary to make the
-documentation clear for users.
-
-```markdown
-# CrankNicolson
-
 !syntax description /Executioner/TimeIntegrator/CrankNicolson
 
-## Overview
+The Crank Nicholson time integration scheme is a second order implicit method. It is unconditionally stable.
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply framework/doc/content/source/timeintegrators/CrankNicolson.md:5]: he s actually a relative of Jack Nicholson

File: framework/doc/content/source/timeintegrators/ExplicitEuler.md [hunk 8]
```diff
@@ -1,33 +1,36 @@
...
+the explicit Euler method can be written:
 
-!syntax inputs /Executioner/TimeIntegrator/ExplicitEuler
+!equation
+U(t+dt) = U(t) + dt A(t, U(t))
 
-!syntax children /Executioner/TimeIntegrator/ExplicitEuler
-```
-!alert-end!
+The Butcher tableau of the quadrature weights for this method is:
```
[the remark this answers is not in the document]
Comment (GiudGiud) [reply framework/doc/content/source/timeintegrators/ExplicitEuler.md:21]: well it s easily google-able. I m trying to not loose documentation by doing what I m doing, but also not compete too hard with wikipedia

File: framework/doc/content/source/userobject/LayeredAverage.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (joshuahansel) [reply framework/doc/content/source/userobject/LayeredAverage.md]: I want a delicious comma after "number"!

File: framework/doc/content/source/timeintegrators/ExplicitEuler.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (joshuahansel) [reply framework/doc/content/source/timeintegrators/ExplicitEuler.md]: That's fine, so I suggest saying something like "This method can be expressed as a Runge-Kutta method with the following Butcher Tableau" (no link necessary).

[the remark this answers is not in the document]
Comment (GiudGiud) [reply framework/doc/content/source/timeintegrators/ExplicitEuler.md]: ok changed

**Comment by GiudGiud** [comment]
Should be good once it passes tests

File: framework/doc/content/source/timeintegrators/AStableDirk4.md [hunk 9]
```diff
@@ -1,33 +1,45 @@
...
 
-!syntax inputs /Executioner/TimeIntegrator/AStableDirk4
+!equation
+\lim_{z->\infty} R(z) = -0.630414937726258
 
-!syntax children /Executioner/TimeIntegrator/AStableDirk4
-```
-!alert-end!
+## Notes:
+- Method is originally due to [!cite](crouzeix1975)
```
Comment (joshuahansel) [inline framework/doc/content/source/timeintegrators/AStableDirk4.md:31]: Needs blank line before

Comment (joshuahansel) [reply framework/doc/content/source/timeintegrators/AStableDirk4.md:31]: And delete the blank line after this item.

File: framework/doc/content/source/timeintegrators/Ralston.md [hunk 10]
```diff
@@ -1,33 +1,50 @@
...
 
-!! Describe and include an example of how to use the Ralston object.
+!equation
+U(t+\Delta t) = U(t) + \dfrac{\Delta t}{4} \left(A(t, U(t)\right) +  \dfrac{3\Delta t}{4} A \left(t + \dfrac{2\Delta t}{3},U(t) + \dfrac{2\Delta t}{3} A(t, U(t)) \right)
 
-!syntax parameters /Executioner/TimeIntegrator/Ralston
+This method can be expressed as a Runge-Kutta method with the following Butcher Tableau:
 
-!syntax inputs /Executioner/TimeIntegrator/Ralston
+!table
```
Comment (joshuahansel) [inline framework/doc/content/source/timeintegrators/Ralston.md:25]: You forgot this Butcher tableau.

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: framework/doc/content/source/timeintegrators/AStableDirk4.md (hunk 9 above)
Comment (joshuahansel) [reply framework/doc/content/source/timeintegrators/AStableDirk4.md:31]: Also it's weird to have a colon after a subsection name

Comment (GiudGiud) [reply framework/doc/content/source/timeintegrators/AStableDirk4.md]: fixed

File: framework/doc/content/source/timeintegrators/Ralston.md (hunk 10 above)
Comment (GiudGiud) [reply framework/doc/content/source/timeintegrators/Ralston.md]: my b

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**